File: 2021/day-1/javascript/solve-puzzle-1.mjs
import { readFileSync } from 'node:fs';

const input = readFileSync('../input.txt', 'utf-8').split('\n');
const len = input.length;

let timesIncreased = 0;

for (let i = 1, previous = Number(input[0]); i < len; i++) {
  const num = Number(input[i]);

  if (num > previous) timesIncreased++;

  previous = num;
}

console.log(timesIncreased);

File: 2021/day-1/python/solve-puzzle-1.py
with open("../input.txt", "r") as handle:
    text = handle.read().splitlines()
    text_len = len(text)
    times_increased = 0
    i = 1

    while i < text_len:
        if int(text[i]) > int(text[i - 1]):
            times_increased += 1

        i += 1

    print(times_increased)
